backend/tests/test_incremental_indexing.py

```
@@ -0,0 +1,189 @@
+import time
+import unittest
+from unittest.mock import patch
+
+import numpy as np
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+
+def setUpModule():
+    from backend import database
+    global original_db_path, temp_module_dir, _saved_faiss_attrs
```

Ruff (0.15.20)

[warning] Using the global statement to update `original_db_path` is discouraged

(PLW0603)